Classes/Console/Command/Database/DatabaseExportCommand.php

```diff
@@ -0,0 +1,190 @@
+<?php
+declare(strict_types=1);
+namespace Helhum\Typo3Console\Command\Database;
+
+/*
+ * This file is part of the TYPO3 Console project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read
+ * LICENSE file that was distributed with this source code.
+ *
+ */
+
+use Helhum\Typo3Console\Command\AbstractConvertedCommand;
+use Helhum\Typo3Console\Database\Configuration\ConnectionConfiguration;
+use Helhum\Typo3Console\Database\Process\MysqlCommand;
+use Helhum\Typo3Console\Database\Schema\TableMatcher;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class DatabaseExportCommand extends AbstractConvertedCommand
+{
+    /**
+     * @var ConnectionConfiguration
+     */
+    private $connectionConfiguration;
+
+    public function __construct(string $name = null, ConnectionConfiguration $connectionConfiguration = null)
+    {
+        parent::__construct($name);
+        $this->connectionConfiguration = $connectionConfiguration ?: new ConnectionConfiguration();
+    }
+
+    protected function configure()
+    {
+        $this->setDescription('Export database to stdout');
+        $this->setHelp(
+            <<<'EOH'
+Export the database (all tables) directly to stdout.
+The mysqldump binary must be available in the path for this command to work.
+This obviously only works when MySQL is used as DBMS.
+
+Tables to be excluded from the export can be specified fully qualified or with wildcards:
+
+<b>Example:</b> <code>%command.full_name% -c Default -e 'cf_*' -e 'cache_*' -e '[bf]e_sessions' -e sys_log</code>
+EOH
+        );
+
+        $this->setDefinition($this->createCompleteInputDefinition());
```

Comment:
Wouldn't it be nicer to use `$this->addOption($name, $shortcut, $mode, $description, $default)` instead of `setDefinition`? It would be less noisy and would remove some of the reasons for creating the `createNativeDefinition` method I can think of.
https://symfony.com/doc/current/console/input.html

Reply:
you could do something like `configureDeprecated` if you want to separate the deprecated options.

Reply:
Yeah, I started that way. However to get the correct help output (which should not include deprecated options/arguments) for all cases including synopsis, I needed to change that to create the definitions instead of setting them directly. So for now, there is no alternative unfortuately.